From 5763dd1c2dd3f1b6e34ad995ca476055fc15b12d Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Mon, 8 Jun 2020 17:29:50 +0800
Subject: [PATCH 6/9] driver: pci: disable regulator when add pcie port failed

When there is no PCIe device connected on board, PCIe controller
driver adds pcie port failed during initialization phase. So, it is
need to disable regulator that is enabled previously.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index f90dcece09c2..bebcadf1787f 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -2825,6 +2825,10 @@ static int imx_pcie_probe(struct platform_device *pdev)
 			} else {
 				dev_err(dev, "unable to add pcie port.\n");
 			}
+
+			if (imx_pcie->epdev_on && regulator_is_enabled(imx_pcie->epdev_on) > 0)
+				ret = regulator_disable(imx_pcie->epdev_on);
+
 			return ret;
 		}
 		/*
-- 
2.17.1

